Automattic/sensei
# Add quiz block structure store and sync

### Changes proposed in this Pull Request

* Add a redux store that tracks quiz structure and updates to and from REST API endpoint
* Add functions that create or update Quiz and Question blocks based on the data structure
* Add functions that extract structure from Quiz and Question block attributes and question description inner blocks content
* ~Enable the lesson's `_lesson_quiz` meta for the lesson rest endpoint, and~ Fix `Sensei_Course_Structure` not creating `_lesson_quiz` meta
* Refactor the store a bit to allow delayed structure saving and retries
* Add unit tests for the store 

### Testing instructions

* Add all types of questions to a lesson, save
* Add Quiz block, 
* Ensure that lesson questions show up, with the correct title, type and description
* Ensure answers for multi-choice, gap fill, true/false questions are also correct

--
* Ensure that the course outline block also still works properly

### TODO
- [x] Add store
- [x] Add block parsing/updating
- [x] Use Quiz post ID instead of lesson ID?
- [x] Use fetch REST API endpoint
- [x] Update data structure to match REST API schema
